### PR TITLE
[10.0] Merge odoo brazil wip/copy views

### PR DESCRIPTION
**Pull request by rvalyi**
see here for the extraction script #529 (comment)

### CODE REVIEW

File: copy_views/models/inherited_ir_ui_view.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    def create(self, values):
+        #
+        # Ensures that the arch of the copied view is realy copied from the
+        # original view
+        #
+        if values.get('copy_id'):
+            copied_view = self.browse(values['copy_id'])
+            values['arch'] = copied_view.read_combined(['arch'])['arch']
+
+        copied_views = []
```
Comment (rvalyi) [inline copy_views/models/inherited_ir_ui_view.py]: useless variable

File: copy_views/models/inherited_ir_ui_view.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+        res = super(IrUiView, self).create(values)
+
+        #
+        # Ensures inheritance gets replicated to copied views
+        #
+        if values.get('inherit_id'):
+            cv = self.search([('copy_id', '=', res.inherit_id.id)])
+            if cv:
```
Comment (rvalyi) [inline copy_views/models/inherited_ir_ui_view.py]: no need for if cv as cv will alway be iterable, for view in cv: is enough

File: copy_views/models/inherited_ir_ui_view.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+                for view in cv:
+                    view.write({'arch': arch})
+
+        return res
+
+    @api.constrains('arch_db')
+    def _check_xml(self):
+        for view in self:
+            #
+            # When the view is being copied, there's no need to validate it's
```
Comment (rvalyi) [inline copy_views/models/inherited_ir_ui_view.py]: it's -> its

File: copy_views/models/inherited_ir_ui_view.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+            cv = self.search([('copy_id', '=', res.inherit_id.id)])
+            if cv:
+                arch = res.read_combined(['arch'])['arch']
+                for view in cv:
+                    view.write({'arch': arch})
+
+        return res
+
+    @api.model
+    def write(self, values):
```
Comment (rvalyi) [inline copy_views/models/inherited_ir_ui_view.py]: should be @api.multi and there are other bugs, seems untested function

**Comment by rvalyi** [comment]
Hello @aricaldeira. At Akretion we liked the idea of this module. I made e few comments and I think the write method is still screwed currently (wrong signature/not tested).

Could you please make a PR with this module against the https://github.com/OCA/server-tools project?

At the moment they would probably block your module because it has no unit test. It would be cool if you could add some... in the OCA/l10n_br_brazil project we could be a bit more tolerant to accept your module before it is accepted into the server-tools project. But at least it would be cool if we could get some review from the big OCA guys before using it here. Not to say it could benefit a lot of people...

You could take some inspiration from this script if you have no clue how to extract the copy_views module and rebase it on the server-tools branch https://github.com/OCA/l10n-brazil/pull/529#issuecomment-305256100


cc @aricaldeira @mileo @renatonlima @pedrobaeza

File: copy_views/models/inherited_ir_ui_view.py [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+            for view_id, cv in copied_views:
+                view = self.browse(view_id)
+                arch = view.read_combined(['arch'])['arch']
+                for view in cv:
+                    view.write({'arch': arch})
+
+        return res
+
+    @api.model
+    def unlink(self):
```
Comment (renatonlima) [inline copy_views/models/inherited_ir_ui_view.py]: should be @api.multi and there are other bugs, seems untested function

**Review by renatonlima — COMMENTED** [review]
Hello guys,

I extract this module and create one PR in server-tools repo https://github.com/OCA/server-tools/pull/863

**Comment by renatonlima** [comment]
As explained in OCA/server-tools#863  the feature of this module already exists in the core, using the primary attribute.

**Comment by rvalyi** [comment]
related to this PR in the WIP branch: https://github.com/odoo-brazil/l10n-brazil-wip/pull/91
Here we see the benefit from interacting more with the core of the OCA to avoid re-inventing the wheel.

**Pull request closed without merging**